lens.cabal

```diff
@@ -192,7 +192,7 @@ library
     indexed-traversable           >= 0.1      && < 0.2,
     indexed-traversable-instances >= 0.1      && < 0.2,
     kan-extensions                >= 5        && < 6,
-    mtl                           >= 2.2.1    && < 2.4,
+    mtl                           >= 2.3      && < 2.4,
```

Comment:
This approach is not going to work, as `mtl` is a boot library that is shipped with GHC. By raising the lower version bounds here, you're cutting off support for GHCs that ship an older version of `mtl` (or, at the very least, making it much harder to construct build plans).

I would advocate for keeping the version bounds as they currently are, but instead guarding the instances behind `#if MIN_VERSION_mtl(2,3,0)` CPP.